addons/hr_maintenance/models/equipment.py

```diff
@@ -1,45 +1,57 @@
 # -*- coding: utf-8 -*-
 
 from odoo import api, fields, models, tools
+from odoo.fields import Domain
 
 
 class MaintenanceEquipment(models.Model):
     _inherit = 'maintenance.equipment'
 
-    employee_id = fields.Many2one('hr.employee', compute='_compute_equipment_assign',
+    employee_id = fields.Many2one('hr.employee', compute='_compute_equipment_assign_to_date',
         store=True, readonly=False, string='Assigned Employee', tracking=True, index='btree_not_null')
-    department_id = fields.Many2one('hr.department', compute='_compute_equipment_assign',
+    department_id = fields.Many2one('hr.department', compute='_compute_equipment_assign_to_date',
         store=True, readonly=False, string='Assigned Department', tracking=True)
     equipment_assign_to = fields.Selection(
-        [('department', 'Department'), ('employee', 'Employee'), ('other', 'Other')],
-        string='Used By',
-        required=True,
-        default='employee')
-    owner_user_id = fields.Many2one(compute='_compute_owner', store=True)
-    assign_date = fields.Date(compute='_compute_equipment_assign', store=True, readonly=False, copy=True)
+        selection_add=[('department', 'Department'), ('employee', 'Employee')], default='employee')
+    is_assigned = fields.Boolean(compute='_compute_is_assigned', search='_search_is_assigned')
 
-    @api.depends('employee_id', 'department_id', 'equipment_assign_to')
+    @api.depends(lambda self: self._get_assign_fields() + ['equipment_assign_to'])
     def _compute_owner(self):
```

Comment:
```suggestion
        @api.depends(lambda self: self._get_assign_fields() + ['equipment_assign_to'])
    def _compute_owner(self):
        other_equipments = self.env[self._name].browse()
        for equipment in self:
            if equipment.equipment_assign_to == 'employee':
                equipment.owner_user_id = equipment.employee_id.user_id.id or False
            elif equipment.equipment_assign_to == 'department':
                equipment.owner_user_id = equipment.department_id.manager_id.user_id.id or False
            elif equipment.equipment_assign_to == 'other':
                equipment.owner_user_id = False
            else:
                other_equipments |= equipment
        if other_equipments:
            super(MaintenanceEquipment, other_equipments)._compute_owner()

```
This keeps the logic in a single loop, only builds one additional recordset, and calls super once